woocommerce/woocommerce-gutenberg-products-block
# Add a Products by Rating filter block

This is an attempt at adding a **Products by Rating** filter block.

For now, I have focused on the core functionality, and there are several pending items on the to-do list:

#### To-do

- [ ] Add support for **Display Options**: Product count, Show 'Apply filters' button, Color
- [ ] Fix the editor preview display
- [ ] Migrate the block from directly using the `Rating` component to utilizing the `CheckboxList` component (this will also make the UX match the preferred option ( pdnLyh-1sR-p2#comment-1044 ))
- [ ] Investigate why the Active Filters block `Clear All` button doesn't work correctly (doesn't clear Rating filters on a Block Shop page)
- [ ] Provide the legacy Filter Products by Rating widget deprecation
- [ ] Add the ability for the `Rating` component to support other contexts (for example the All Products block)

### Screenshots
![New_shop_–_ratings](https://user-images.githubusercontent.com/905781/188260420-cff10af5-9ed1-45ea-a700-b5dfd07f5ed4.jpg)

### Testing
- Activate a block theme & setup WooCommerce with products that have some ratings
- Create a classic shop page and a separate page using the `All Products` block
- Add the `Filter Products by Rating` block and `Active Filters` block on both pages
- Check that `Filter Products by Rating` block is correctly:
  - Displaying the star rating buttons within the filter block
  - Displaying the active state (stars using the Primary color and the product count are in bold)
  - Filtering the products in the product loop (on both Classic and block shop pages)
  - Clearing after getting removed via `Active Filters` remove button
  - Updating the URL with the rating parameter
  - Loading the rating filters via URL parameters on page reload

#### User Facing Testing

Same as the above steps

### WooCommerce Visibility




### Changelog

> Add Filter Products by Rating block
